src/segment-loader.js

```diff
@@ -2368,6 +2368,16 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     this.logger_(`Requesting ${segmentInfoString(segmentInfo)}`);
 
+    // if init segment is being requested
+    // we will have to re-append the init segment.
+    if (simpleSegment.map && !simpleSegment.map.bytes) {
+      this.logger_('going to request init segment.');
+      this.appendInitSegment_ = {
+        video: true,
+        audio: true
+      };
```

Comment:
Should this be dependent on the type that this loader is handling?

Reply:
I don't think it should because if we get an audio or video init segment on this segment loader, we should re-append it. It won't likely happen for a video only loader or an audio only loader, but at this point we won't actually know what type of content we are requesting, as we get that in trackinfo.